holidays/holiday_base.py

```diff
@@ -746,14 +749,14 @@ def _add_special_holidays(self, mapping_names, observed=False):
                     )
                 else:  # Substituted holidays.
                     to_month, to_day, from_month, from_day, *optional = data
+                    from_date = date(optional[0] if optional else self._year, from_month, from_day)
                     self._add_holiday(
                         self.tr(self.substituted_label)
-                        % date(
-                            optional[0] if optional else self._year, from_month, from_day
-                        ).strftime(self.tr(self.substituted_date_format)),
+                        % from_date.strftime(self.tr(self.substituted_date_format)),
                         to_month,
                         to_day,
                     )
+                    self.workdays.add(from_date)
```

Comment:
I'm not sure if it's worth it to put an explicit _is_weekend check before adding date into weekend_workdays.
```suggestion
                    self.weekend_workdays.add(from_date)
```

Reply:
I think such a check would be a bit redundant. 👀